### ISSUE TITLE
(Not a bug) "test_build.test_image_glob" fails on Windows

### ISSUE THREAD

**Issue by hhsprings** [sphinx-doc/sphinx#3015]
This is simply path separator problem.

```
FAIL: test_build.test_image_glob
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\Python35\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "c:\Users\hhsprings\work\sphinx\tests\util.py", line 278, in deco
    func(app, status, warning, *args2, **kwargs2)
  File "c:\Users\hhsprings\work\sphinx\tests\test_build.py", line 151, in test_image_glob
    assert doctree[0][1]['candidates'] == {'*': 'subdir/rimg.png'}
AssertionError: 
    assert isinstance(<document: <section "test-image-glob/subdir"...>>[0][1], <module 'docutils.nodes' from 'c:\\Python35\\lib\\site-packages\\docutils\\nodes.py'>.image)
>>  assert <document: <section "test-image-glob/subdir"...>>[0][1]['candidates'] == {'*': 'subdir/rimg.png'}
    assert <document: <section "test-image-glob/subdir"...>>[0][1]['uri'] == 'subdir/rimg.png'
```

``` python
@with_app(buildername='dummy', testroot='image-glob')
def test_image_glob(app, status, warning):
    app.builder.build_all()

    # index.rst
    doctree = pickle.loads((app.doctreedir / 'index.doctree').bytes())

    assert isinstance(doctree[0][1], nodes.image)
    assert doctree[0][1]['candidates'] == {'*': 'rimg.png'}
    assert doctree[0][1]['uri'] == 'rimg.png'

    assert isinstance(doctree[0][2], nodes.figure)
    assert isinstance(doctree[0][2][0], nodes.image)
    assert doctree[0][2][0]['candidates'] == {'*': 'rimg.png'}
    assert doctree[0][2][0]['uri'] == 'rimg.png'

    assert isinstance(doctree[0][3], nodes.image)
    assert doctree[0][3]['candidates'] == {'application/pdf': 'img.pdf',
                                           'image/gif': 'img.gif',
                                           'image/png': 'img.png'}
    assert doctree[0][3]['uri'] == 'img.*'

    assert isinstance(doctree[0][4], nodes.figure)
    assert isinstance(doctree[0][4][0], nodes.image)
    assert doctree[0][4][0]['candidates'] == {'application/pdf': 'img.pdf',
                                              'image/gif': 'img.gif',
                                              'image/png': 'img.png'}
    assert doctree[0][4][0]['uri'] == 'img.*'

    # subdir/index.rst
    doctree = pickle.loads((app.doctreedir / 'subdir/index.doctree').bytes())

    from os import path  ###
    assert isinstance(doctree[0][1], nodes.image)
    assert doctree[0][1]['candidates'] == {'*': 'subdir' + path.sep + 'rimg.png'}  ###
    assert doctree[0][1]['uri'] == 'subdir' + path.sep + 'rimg.png'  ###

    assert isinstance(doctree[0][2], nodes.image)
    assert doctree[0][2]['candidates'] == {'application/pdf': 'subdir/svgimg.pdf',
                                           'image/svg+xml': 'subdir/svgimg.svg'}
    assert doctree[0][2]['uri'] == 'subdir' + path.sep + 'svgimg.*'  ###

    assert isinstance(doctree[0][3], nodes.figure)
    assert isinstance(doctree[0][3][0], nodes.image)
    assert doctree[0][3][0]['candidates'] == {'application/pdf': 'subdir/svgimg.pdf',
                                              'image/svg+xml': 'subdir/svgimg.svg'}
    assert doctree[0][3][0]['uri'] == 'subdir' + path.sep + 'svgimg.*'  ###
```

<s>
I don't think this should be fixed, but if it was fixed, maybe contributors working on Windows would be happier.
</s>
If this result is affecting to html output, it is the case, if not it's okey. I'll check it.

<hr/>

No problem. The output html doesn't have `\` as url separator. So again,

> I don't think this should be fixed, but if it was fixed, maybe contributors working on Windows would be happier.

**Comment by hhsprings** [comment]
Maybe `test_build_html.test_html_warnings`, `test_build_latex.test_latex_warnings`, and `test_build_texinfo.test_texinfo_warnings` has the same problem.

`test_ext_intersphinx.test_missing_reference` also fails on my environment, but I can't say it is the same issue or not, yet. (I'm going to investigate it.)

**Comment by hhsprings** [comment]
The error of `test_build_html.test_html_warnings` (and also `test_build_latex.test_latex_warnings`, `test_build_texinfo.test_texinfo_warnings`) is not the same as of `test_build.test_image_glob` though it is also path separator problem. See:

```
WARNING: Encoding 'utf-8-sig' used for reading included file 'c://Users//hhsprings//work//sphinx//tests//build//test-warnings//wrongenc.inc' seems to be wrong, try giving an :encoding: option
```

So, I'll separate this into another ticket. -> #3016

**Comment by hhsprings** [comment]
`test_ext_intersphinx.test_missing_reference`  is the same as of `test_build.test_image_glob`.

``` python
    from os import path
    rn = reference_check('py', 'mod', 'py3krel:module1', 'foo')
    assert rn['refuri'].replace(path.sep, "/") == 'py3k/foo.html#module-module1'

    rn = reference_check('py', 'mod', 'py3krelparent:module1', 'foo')
    assert rn['refuri'].replace(path.sep, "/") == '../../py3k/foo.html#module-module1'

    rn = reference_check('py', 'mod', 'py3krel:module1', 'foo', refdoc='sub/dir/test')
    assert rn['refuri'].replace(path.sep, "/") == '../../py3k/foo.html#module-module1'

    rn = reference_check('py', 'mod', 'py3krelparent:module1', 'foo', refdoc='sub/dir/test')
    assert rn['refuri'].replace(path.sep, "/") == '../../../../py3k/foo.html#module-module1'
```

**Comment by tk0miya** [comment]
Did you mean `refuri` includes `os.path.sep` character?
AFAIK, `refuri` and `uri` should use `/` as path separators. It might be a bug of building the `refuri`.

@shimizukawa Could you check this please?

**Comment by shimizukawa** [comment]
Fixed at 76f2ed9. Thanks for your reporting.
